digdag-core/src/main/java/io/digdag/core/agent/RequireOperatorFactory.java

```diff
@@ -64,46 +68,109 @@ private RequireOperator(OperatorContext context, TaskCallbackApi callback)
+            }
+            catch (ResourceNotFoundException ex) {
+                throw new TaskExecutionException(String.format(ENGLISH, "Dependent workflow does not exist. %s, workflowName:%s",
+                        projectIdentifier.transform((p)->p.toString()).or(""), workflowName ));
+            }
+            catch (ResourceLimitExceededException ex) {
+                throw new TaskExecutionException(ex);
+            }
 
-                boolean isDone = attempt.getStateFlags().isDone();
-                if (isDone) {
-                    if (!ignoreFailure && !attempt.getStateFlags().isSuccess()) {
+            if (sessionAttemptConflictException.isPresent()) {
+                StoredSessionAttempt conflictedAttempt = sessionAttemptConflictException.get().getConflictedSession();
+                if (conflictedAttempt.getStateFlags().isDone()) {
+                    // A flag to distinguish whether the attempt is kicked by require> or previous attempt.
+                    boolean requireKicked = lastStateParams.get("require_kicked", boolean.class, false);
+                    if (!requireKicked &&  (
+                            rerunOn == OptionRerunOn.ALL ||
+                            rerunOn == OptionRerunOn.FAILED && !conflictedAttempt.getStateFlags().isSuccess())) {
+
+                        //To force run, set flag gen_retry_attempt_name and do polling
+                        throw nextPolling(lastStateParams.deepCopy().set("rerun_on_retry_attempt_name", UUID.randomUUID().toString()));
+                    }
+                    if (!ignoreFailure && !conflictedAttempt.getStateFlags().isSuccess()) {
                         // ignore_failure is false and the attempt is in error state. Make this operator failed.
                         throw new TaskExecutionException(String.format(ENGLISH,
-                                    "Dependent workflow failed. Session id: %d, attempt id: %d",
-                                    attempt.getSessionId(), attempt.getId()));
+                            "Dependent workflow failed. Session id: %d, attempt id: %d",
+                            conflictedAttempt.getSessionId(), conflictedAttempt.getId()));
                     }
                     return TaskResult.empty(cf);
                 }
                 else {
-                    // TODO use exponential-backoff to calculate retry interval
-                    throw TaskExecutionException.ofNextPolling(1, ConfigElement.copyOf(request.getLastStateParams()));
+                    // Wait for finish running attempt
+                    throw nextPolling(request.getLastStateParams().deepCopy());
                 }
             }
-            catch (ResourceNotFoundException ex) {
-                throw new ConfigException(ex);
+            else if (attempt.isPresent()) { // startSession succeeded and created new attempt
+                throw nextPolling(request.getLastStateParams().deepCopy().set("require_kicked", true));
```

Comment:
How about moving this line to right after calling `callback.startSession()`  in the above `try { ... }` block? Maybe we can remove the Optional?

Reply:
My intention is same as above. But this code is small. 
So I agree with move to `try{..}`.

Reply:
Fixed it. Thanks!